crates/walrus-sui/src/client/retry_client.rs

```diff
@@ -1400,6 +1409,32 @@ impl RetriableRpcClient {
         self.client.with_failover(request).await
     }
 
+    /// Records RPC latency metrics
+    fn record_rpc_latency(&self, method: &str, status: &str, duration: Duration) {
+        if let Some(metrics) = &self.metrics {
+            metrics
+                .rpc_latency
+                .with_label_values(&[method, self.client.get_current_client_name(), status])
```

Comment:
nit: in Walrus repo, we use `walrus_utils::with_label!` macro in most of places, can we keep it consistent here?